### ISSUE TITLE
Some specified versions not vivified

### ISSUE THREAD

**Issue by gremble** [jamesmcdonald/puppet-cpanm#13]
I'm make the following package requests via your module:

"Archive::Tar@2.40"
"YAML@1.30"
"Plack@1.0050"

But Puppet does not upgrade them. For example I can run:

`cpanm Archive::Tar@2.38`

On the command line as root to install the 2.38 version. A puppet which has Archive::Tar@2.40 as the cpanm package name, does not cause and upgrade, the version sits at 2.38. I've no idea what could be going on. Any ideas?

**Comment by jamesmcdonald** [comment]
There's an error in the version check from an earlier PR that gets upset when things end in 0. I'll try to have a look this evening.

**Comment by jamesmcdonald** [comment]
I've made a PR that will hopefully fix it. Are you able to try it in your environment? I don't have a lot of Perl stuff these days so it would be cool to get feedback and check it doesn't break everything!

**Comment by gremble** [comment]
I'll give it a try over the next couple of days.

**Comment by gremble** [comment]
Right then, basically all good, but some things happened along the way.

Pre-fix module failed to downgrade correctly (in addition to the reported upgrade issue):
   Image::Size    PUPPET: 3.300   LOCAL: 3.3
   Perl::Critic    PUPPET: 1.140   LOCAL: 1.150
   Perl::Critic::Utils    PUPPET: 1.140   LOCAL: 1.150
   Throwable::Error    PUPPET: 1.000   LOCAL: 1.001
   Archive::Tar    PUPPET: 2.30   LOCAL: 2.40
   Date::Parse    PUPPET: 2.30   LOCAL: 2.33
   HTML::Template::Pro    PUPPET: 0.9510   LOCAL: 0.9524
   Module::CoreList    PUPPET: 5.20181130   LOCAL: 5.20230120
   Text::CSV_XS    PUPPET: 1.40   LOCAL: 1.50
   Dancer::Request::Upload    PUPPET: 1.3520   LOCAL: 1.3521


On the fix branch it's all basically fixed, in both upgrade and downgrade directions but I needed to make a change:

I was defining "Plack::Handler::Starman@undef" which fails like this:

```
! Finding Plack::Handler::Starman@undef on cpanmetadb failed.
! Finding Plack::Handler::Starman@undef () on mirror http://www.cpan.org failed.
! Couldn't find module or a distribution Plack::Handler::Starman@undef
Error: 
Error: /Stage[main]/Aperture::Serversoftware/Cpanm[Plack::Handler::Starman@undef]/ensure: change from 'absent' to 'present' failed:  (corrective)
```

With undef removed it works perfectly. I've no idea how undef ended up in there and I'm glad the new module version caught it.




There's an issue which comes up with Image::Magick though it still installs. The version on CPAN is 6.9.12-2 (or the older 6.9.12-1) so we have another quirk of CPAN repository versioning here. Anyhow it still installs. What an annoying corner case. I can't specifiy 6.9.12-2 in the cpanm (or cpan) command so it's not a fix we can make I don't think.

```
! Finding Image::Magick (== 6.9.12) on cpanmetadb failed.
Notice: /Stage[main]/Aperture::Serversoftware/Cpanm[Image::Magick@6.9.12]/ensure: created
```


Thanks so much for this fix.

**Comment by jamesmcdonald** [comment]
That's merged and pushed to the forge as 0.9.16.
